ocaml/xapi/taskHelper.ml

```diff
@@ -178,22 +178,26 @@ let status_is_completed task_status =
   task_status = `success || task_status = `failure || task_status = `cancelled
 
 let complete ~__context result =
-  Context.complete_tracing __context ;
-  operate_on_db_task ~__context (fun self ->
-      let status = Db_actions.DB_Action.Task.get_status ~__context ~self in
-      if status = `pending then (
-        Db_actions.DB_Action.Task.set_allowed_operations ~__context ~self
-          ~value:[] ;
-        Db_actions.DB_Action.Task.set_finished ~__context ~self
-          ~value:(Date.of_float (Unix.time ())) ;
-        Db_actions.DB_Action.Task.set_progress ~__context ~self ~value:1. ;
-        set_result_on_task ~__context self result ;
-        Db_actions.DB_Action.Task.set_status ~__context ~self ~value:`success
-      ) else
-        debug "the status of %s is: %s; cannot set it to `success"
-          (Ref.really_pretty_and_small self)
-          (status_to_string status)
-  )
+  Xapi_stdext_pervasives.Pervasiveext.finally
```

Comment:
These diffs can be a bit difficult to read because of the change indendation, a way to leave it the same as before would be to add a function:

```ocaml
let finally_complete_tracing ~__context f =
  Xapi_stdext_pervasives.Pervasiveext.finally
    f (fun () -> Context.complete_tracing __context)
```

Then use it like:
```ocaml 
  let@ () = finally_complete_tracing ~__context in
  operate_on_db_task ~__context (fun self -> ...
```
or using the `@@` operator:
```ocaml 
  finally_complete_tracing ~__context @@ fun () ->
  operate_on_db_task ~__context (fun self -> ...
```

Reply:
Ah, I wasn't ignoring whitespace -_-

Reply:
https://github.com/xapi-project/xen-api/pull/5735/commits/7c0d3da5f53caec3d8e34994a5f3c700c73a1128 Done!